in_toto/runlib.py

```diff
@@ -82,6 +92,14 @@ def record_artifacts_as_dict(artifacts):
     The files that result form a link command execution are called
     products.
 
+    Excludes files that are matched by the file patterns specified in
+    ARTIFACT_EXCLUDES setting.
+    Note: Exclude patterns are applied on the basename, i.e. the pattern
+    "foo" excludes /foo as well as /subdir/foo
```

Comment:
Actually it is not a consequence of using fnmatch. I implemented this behavior on purpose, because I thought this is how .gitignore works. But maybe I was wrong. Do you want me to change this now?